src/components/Link.jsx

```diff
@@ -18,9 +18,10 @@ const Link = ({
   const handleClick = (e) => {
     e.preventDefault();
 
-    if (typeof onClick === 'function') {
```

Comment:
@JForsaken is this line a problem for you? If yes we can bump this to a major.

Reply:
it was only to sanitize if `onClick` was passed something other than a function, to prevent `Uncaught TypeError: onClick is not a function`

Reply:
If you want to remove this check, I'm fine with it and I don't think it needs to be a major if so 👍

Reply:
Yeah gotcha, we discussed it and I think this shouldn't be an issue.

* PropTypes is set to `func`, so one of the unit test output a proptype warning even if everything passes.
* I'm not sure if we should double check for every possible type of every prop in the components... that defetas the use of proptypes..
* You shouldn't pass anything else than a function in a prop called "onClick" in the first place :p

So unless you have an actual onClick as a string in one of your apps, I'll keep the `3.x` tag

Reply:
Good with me! And yeah `3.x` tag is good 💪
And no, I don't have an `onClick` set as a `String` in one of my apps, I prefer them being `Number`s
Just kidding merge it 💯